### ISSUE TITLE
[Bug] Error when launching GRASS GIS 8.0.0-RC1-1 on Windows

### ISSUE THREAD

**Issue by isaacnewtonfx** [OSGeo/grass#2101]
**Describe the bug**
After installing GRASS GIS 8.0.0-RC1-1, it failed to launch on my laptop with a strange error message.

**To Reproduce**

1. Download GRASS 8.0.0-RC1-1 from [grass gis website](https://grass.osgeo.org/grass80/binary/mswindows/native/WinGRASS-8.0.0RC1-1-Setup.exe)
2. Install the program on windows 10
3. After installation, launch the program

**Expected behavior**
It has to open the GRASS GIS 8.0.0 graphical user interface without an error

**Screenshots**
![image](https://user-images.githubusercontent.com/8677269/149738751-f5283845-6b8b-47d1-9181-38f88ed2b9dd.png)

**System description (please complete the following information):**

- Operating System:  Windows 10 Entreprise version 1909
- GRASS GIS version 8.0.0 RC1-1

**Additional context**
I installed it using administrator rights and launched it using administrator rights.
The Windows user is also an Administrator.

**Comment by nilason** [comment]
This has been addressed for RC2 (https://github.com/OSGeo/grass/commit/04af08cd45e48f4b7016366bc29826bd40317dce), remains to see if also solved (see https://lists.osgeo.org/pipermail/grass-dev/2022-January/095519.html).

**Comment by isaacnewtonfx** [comment]
That is great @nilason. I would try the RC2 when the windows executable is ready and give feedback on the issue.

**Comment by nilason** [comment]
https://grass.osgeo.org/grass80/binary/mswindows/native/WinGRASS-8.0.0RC2-1-Setup.exe

**Comment by isaacnewtonfx** [comment]
Wow the executable is now available...let me test it at work today and give feedback.

**Comment by isaacnewtonfx** [comment]
I have downloaded the RC2 setup. It runs into the same error.

![image](https://user-images.githubusercontent.com/8677269/149900706-8e310bb6-6c9d-419d-b0f7-1f13c0c5c57d.png)

![image](https://user-images.githubusercontent.com/8677269/149900935-5729ad5a-7d89-4dd1-abb8-30c8f7b43411.png)

**Comment by petrasovaa** [comment]
Do you have non-english locale?
See also https://github.com/wxWidgets/Phoenix/issues/1751, it's a mess, likely related to Python 3.8

**Comment by isaacnewtonfx** [comment]
No, everything is set to English.

![image](https://user-images.githubusercontent.com/8677269/149974660-e2c603e3-dce7-426a-a378-fb892824d393.png)

![image](https://user-images.githubusercontent.com/8677269/149974782-21df3af6-340a-478e-a83d-16784d9dd00e.png)

**Comment by petrasovaa** [comment]
What about the Regional format? Could you change that and try again? Not sure if restart is required.

**Comment by isaacnewtonfx** [comment]
Ok, I will try that and get back.

**Comment by nilason** [comment]
> This has been addressed for RC2 ([04af08c](https://github.com/OSGeo/grass/commit/04af08cd45e48f4b7016366bc29826bd40317dce)), remains to see if also solved (see https://lists.osgeo.org/pipermail/grass-dev/2022-January/095519.html).

Looking closer at the screenshot, I realise my previously linked commit and mail list post are not related to this reported issue.
I just successfully installed and started RC2 on a Windows 10 Pro machine.

**Comment by isaacnewtonfx** [comment]
The changes to the locale settings on my Windows 10 enterprise did not make any difference.

**Comment by nilason** [comment]
For better overview, this is the error message in text:

```
C: \Users\Newton>Traceback (most recent call last):
  File "C: \Program Files\GRASS GIS 8.0\gui\wxpython\wxgui.py", line 72, in OnInit
    introImage = wx.Image (introImagePath, wx.BITMAP_TYPE_PNG)
wx-_core.wxAssertionError: (++ assertion "stremp(setlocale(0,0), "C") == 0" failed at
D:\srclosge04w|src(wxwidgets\wxwidgets-3.1.5-Odf1d81acd6f1be8624022f8eecb51679008ca40\src\common\intl.cpp(1694)
in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is
a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!
OnInit returned false, exiting...
```

**Comment by neteler** [comment]
Could something like this help (wild guess by searching around):

- "run `self.locale = wx.Locale(wx.LANGUAGE_ENGLISH)` as part of ..." ([source](https://github.com/CellProfiler/CellProfiler/issues/4220#issuecomment-689895071))

or, modified:
- `locale.setlocale(locale.LC_ALL, wx.GetLocale().Name)`

for full code, see
https://github.com/CellProfiler/CellProfiler/commit/8525f654af683150d9123fd99d8eccdb531eaf0c

**Comment by petrasovaa** [comment]
Possibly, not quite sure what the implications of that are. This is the project's latest code:
https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/gui/app.py#L60

Obviously this needs to be well tested.

**Comment by hellik** [comment]
Are there more Python versions in %PATH% ?

**Comment by nilason** [comment]
https://github.com/wxWidgets/wxWidgets/blob/9c0a8be1dc32063d91ed1901fd5fcd54f4f955a1/src/common/intl.cpp#L1699

**Comment by nilason** [comment]
Triggering code in wxWidgets has changed in master:
https://github.com/wxWidgets/wxWidgets/commit/c9e2143a7a97ef82bcc20e8c6956663523a62adf
commit message may give some clue.

**Comment by hellik** [comment]
it seems to be a Windows CRT related issue. 

see  https://github.com/OSGeo/grass/discussions/2064

**Comment by isaacnewtonfx** [comment]
1. Regarding the python paths available on Environment Variables %PATH%, I have ArcGIS 10.8 and ArcGIS Pro. 
2. ArcGIS 10.8 is running on Python 2.7 and ArcGIS Pro is running on python 3.7.10. 
3. The ArcGIS 10.8 Python 2.7 is what gets recognized when called in Windows Command Prompt by default

![image](https://user-images.githubusercontent.com/8677269/150067474-86f755fc-2d3a-4d09-96d4-2bc591efbd15.png)

![image](https://user-images.githubusercontent.com/8677269/150067963-58b90448-7a30-431a-8ad9-387ba2ae18d9.png)

**Comment by hellik** [comment]
Could you rename temporarily step by by these Arcgis pythons and try to start winGRASS?

**Comment by isaacnewtonfx** [comment]
> Could you rename temporarily step by by these Arcgis pythons and try to start winGRASS?

Just tried that, but did not make a difference. I want to try Grass 8.0.0 RC2 on a machine that has a fresh Windows installation to see how it works there and I would give the feedback.

**Comment by isaacnewtonfx** [comment]
GASS 8.0.0 RC2 is running without a problem on a freshly installed Windows 10 Pro. Could it be a Windows 10 Enterprise specific problem?

See below, specs of the laptop and freshly installed Windows 10 Pro
![laptop specs](https://user-images.githubusercontent.com/8677269/150138224-6a2dcb56-963f-4d85-9383-934acbf1049a.png)

The Locale information on this windows 10 pro
![image](https://user-images.githubusercontent.com/8677269/150138472-c0422e7e-adb5-48da-9d47-3a7e510c2322.png)
![image](https://user-images.githubusercontent.com/8677269/150138587-133a165d-82ec-4d98-a4e7-bba9dc755e56.png)

Below is the GRASS 8.0.0 interface
![image](https://user-images.githubusercontent.com/8677269/150138295-4cb19a3e-8876-4f36-a383-7f194281c3ce.png)

**Comment by hellik** [comment]
> > Could you rename temporarily step by by these Arcgis pythons and try to start winGRASS?
> 
> Just tried that, but did not make a difference.

have you installed the MS runtimes provided by the winGRASS installer?

**Comment by hellik** [comment]
>Could it be a Windows 10 Enterprise specific problem?

it seems so.  it may be some kind of a CRT issue, as Microsoft is moving away from MSCRT to forced use of UCRT.

**Comment by isaacnewtonfx** [comment]
> it seems so. it may be some kind of a CRT issue, as Microsoft is moving away from MSCRT to forced use of UCRT.

Yes, I do check the box to install the CRT that comes with the grass installer. I reinstalled it but still the same old problem. I also turned off completely the Windows Security app but still old problem. Quite a funny thing going on somewhere...

**Comment by isaacnewtonfx** [comment]
> Could something like this help (wild guess by searching around):
> 
> * "run `self.locale = wx.Locale(wx.LANGUAGE_ENGLISH)` as part of ..." ([source](https://github.com/CellProfiler/CellProfiler/issues/4220#issuecomment-689895071))

@neteler 

I tried Grass 8.0.0 RC2 On another desktop running Windows 10 Pro version 21H1 and the same problem occurred.

I edited the file at the "Program Files\GRASS GIS 8.0\gui\wxpython\wxgui.py" as shown below and the error went away. See below changes. However, I tried these changes on Windows 10 Enterprise and it did not solve the problem.

![changes to wxgui](https://user-images.githubusercontent.com/8677269/150316749-974f7998-0c67-44a0-ad94-4b134e74c274.PNG)

Details of the Windows 10 Pro Desktop desktop on which the changes were done is as shown below.
![details](https://user-images.githubusercontent.com/8677269/150316900-a231068b-a22e-4c6f-bdf4-3efa17b4c4bd.PNG)

**Comment by petrasovaa** [comment]
You can try the code I [linked](https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/gui/app.py#L60):
```
            import locale
            # Need to startup wx in English, otherwise C++ can't load images.
            self.locale = wx.Locale(wx.LANGUAGE_ENGLISH)
            # Ensure Python uses the same locale as wx
            locale.setlocale(locale.LC_ALL, self.locale.GetName())
```

**Comment by isaacnewtonfx** [comment]
> import locale
>             # Need to startup wx in English, otherwise C++ can't load images.
>             self.locale = wx.Locale(wx.LANGUAGE_ENGLISH)
>             # Ensure Python uses the same locale as wx
>             locale.setlocale(locale.LC_ALL, self.locale.GetName())

Just tried it, the locale error did not come up. But now a different error (see below)

![Capture1](https://user-images.githubusercontent.com/8677269/150483283-78555fe4-9568-4054-a74f-18dfadd9aad9.PNG)

![Capture2](https://user-images.githubusercontent.com/8677269/150483339-7ff8d3f7-c215-4e75-9d94-5671e5dc9492.PNG)

**Comment by petrasovaa** [comment]
I merged a fix, the GUI should at least start now, but the problem with loading libraries would be still there (not sure why that happens), so some functionality wouldn't work. You would need to download a new version https://wingrass.fsv.cvut.cz/grass80/ to test.

**Comment by isaacnewtonfx** [comment]
Well noted, I would wait on a new version to test it and revert on how it goes.

**Comment by nCVJTK** [comment]
Hello...has anyone been able to fix wxlocale error upon starting grass gis:

![image](https://user-images.githubusercontent.com/57320462/152440211-91c5efd1-30e0-4ffc-ab80-e624acc81258.png)

The same error i get with grass 7.8.6 and new 8.0.0...my pc runs on win 10 pro....thx!

**Comment by landam** [comment]
Another two Windows builds affected by this bug:

![image (6)](https://user-images.githubusercontent.com/5683186/155320437-0a0f859f-c349-4ffb-97fc-4ec119606ce2.png)

![AboutSystems_Nangula1](https://user-images.githubusercontent.com/5683186/155320540-19328d21-f573-4f7f-9abd-131a8cede6f8.JPG)

**Comment by petrasovaa** [comment]
There is a chance this will go away with the new wxPython 4.2.0, see https://github.com/wxWidgets/Phoenix/issues/1637

**Comment by landam** [comment]
@isaacnewtonfx Please try out recent 8.3 from https://wingrass.fsv.cvut.cz/grass83/ It comes with wxPython 4.2.0...

![VirtualBox_MS Windows_23_09_2022_12_04_27](https://user-images.githubusercontent.com/5683186/191939715-90c9e262-ee05-414a-bb8a-4df106b2b3d7.png)

**Comment by isaacnewtonfx** [comment]
Alright, i will give it a try

**Comment by girishnand** [comment]
I'm still getting this error on startup even with GRASS GIS 8.2.0 - 

```
Starting GRASS GIS...
WARNING: Concurrent mapset locking is not supported on Windows
Cleaning up temporary files...

          __________  ___   __________    _______________
         / ____/ __ \/   | / ___/ ___/   / ____/  _/ ___/
        / / __/ /_/ / /| | \__ \\_  \   / / __ / / \__ \
       / /_/ / _, _/ ___ |___/ /__/ /  / /_/ // / ___/ /
       \____/_/ |_/_/  |_/____/____/   \____/___//____/

Welcome to GRASS GIS 8.2.0
GRASS GIS homepage:                      https://grass.osgeo.org
This version running through:            Command Prompt (C:\Windows\system32\cmd.exe)
Help is available with the command:      g.manual -i
See the licence terms with:              g.version -c
See citation options with:               g.version -x
If required, restart the GUI with:       g.gui wxpython
When ready to quit enter:                exit

Launching <wxpython> GUI in the background, please wait...
Microsoft Windows [Version 10.0.19044.2486]
(c) Microsoft Corporation. All rights reserved.

C:\Users\Girish>Traceback (most recent call last):
  File "C:\Program Files\GRASS GIS 8.2\gui\wxpython\wxgui.py", line 73, in OnInit
    introImage = wx.Image(introImagePath, wx.BITMAP_TYPE_PNG)
wx._core.wxAssertionError: C++ assertion "strcmp(setlocale(0, 0), "C") == 0" failed at D:\src\osgeo4w\src\wxwidgets\wxwidgets-3.1.5-0df1d81acd6f1be8624022f8eecb51679008ca40\src\common\intl.cpp(1694) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!
OnInit returned false, exiting...

```


I am on windows 10 Home. 

My computer's "Region Format" is set to "English (India)". If I change this to "English (United States)" from the control panel, the error goes away, and GRASS GIS starts. But the error comes right back if I change the region format back.

**Comment by neteler** [comment]
Please try the new GRASS GIS 8.2.1:
https://grass.osgeo.org/news/2023_01_21_grass_gis_8_2_1_released/

**Comment by girishnand** [comment]
> Please try the new GRASS GIS 8.2.1: https://grass.osgeo.org/news/2023_01_21_grass_gis_8_2_1_released/

The bug is fixed with 8.2.1. 
Thanks muchly.

**Comment by echoix** [comment]
@isaacnewtonfx is your problem still relevant or not fixed? The second problem by another user @girishnand was resolved with 8.3.0

**Comment by isaacnewtonfx** [comment]
@echoix  thanks for your concern. I am now running Grass GIS 8.3.1. It is able to launch the GUI though i see an error message in the console (See attached) but it does not prevent me from using the software.

![Capture](https://github.com/OSGeo/grass/assets/8677269/aa1b1ca3-ac43-42d2-82a2-aeb04570fde6)

**Comment by echoix** [comment]
Ok, seems like a different issue. We can close this, and we can create a new issue from your last comment.
